monkey/monkey_island/cc/resources/agent_heartbeat.py

```diff
@@ -0,0 +1,33 @@
+from http import HTTPStatus
+from json import JSONDecodeError
+
+from flask import request
+
+from common.types import AgentID
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+
+
+class AgentHeartbeat(AbstractResource):
+    urls = ["/api/agent/<uuid:agent_id>/heartbeat"]
+
+    def __init__(self, island_event_queue: IIslandEventQueue):
+        self._island_event_queue = island_event_queue
+
+    # Used by the agent. Can't secure.
+    def post(self, agent_id: AgentID):
+        try:
+            heartbeat_timestamp = request.json["heartbeat_timestamp"]
+            if not (isinstance(heartbeat_timestamp, int) or isinstance(heartbeat_timestamp, float)):
+                raise TypeError("Terminate signal's timestamp is not a number")
+            elif heartbeat_timestamp <= 0:
+                raise ValueError("Terminate signal's timestamp is not a positive number")
```

Comment:
Rather than doing custom validation, I'd suggest creating a pydantic object with validation rules and passing it around. Otherwise we'll have to repeat these ifs all over the place

Reply:
I'll make another PR to do that here and in `terminate_all_agents.py`.